lib/src/main/java/tech/relaycorp/relaydroid/persistence/EncryptedDiskPersistence.kt

```diff
@@ -0,0 +1,84 @@
+package tech.relaycorp.relaydroid.persistence
+
+import android.content.Context
+import androidx.security.crypto.EncryptedFile
+import androidx.security.crypto.MasterKey
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import tech.relaycorp.relaydroid.common.Logging.logger
+import java.io.File
+import java.io.IOException
+import java.util.logging.Level
+import kotlin.coroutines.CoroutineContext
+
+class EncryptedDiskPersistence(
+    private val context: Context,
+    private val coroutineContext: CoroutineContext = Dispatchers.IO,
+    private val rootFolder: String = "relaydroid"
+) : Persistence {
+
+    @Suppress("BlockingMethodInNonBlockingContext")
+    override suspend fun set(location: String, data: ByteArray) {
+        withContext(coroutineContext) {
+            delete(location)
+            buildEncryptedFile(location)
+                .openFileOutput()
+                .use { it.write(data) }
+        }
+    }
+
+    @Suppress("BlockingMethodInNonBlockingContext")
+    override suspend fun get(location: String): ByteArray? = withContext(coroutineContext) {
+        try {
+            buildEncryptedFile(location)
+                .openFileInput()
+                .use { it.readBytes() }
+        } catch (exception: IOException) {
```

Comment:
Separate note: I think we need a consistent way to deal with `IOException` in all the methods in this class. I think that's another reason to use a custom class like `Persistence[Error|Exception]` -- For example, we could throw the following when we fail to write to a file:

```kotlin
try {
        withContext(coroutineContext) {
            delete(location)
            try {
                buildEncryptedFile(location)
                    .openFileOutput()
                    .use { it.write(data) }
            } catch (exception: IOException) {
                  throw PersistenceException(exception, "Failed to write to file $location")
            }
        }

```

... Which will provide better context for what went wrong.